test/examples/examples.test.ts

```diff
@@ -0,0 +1,109 @@
+import { createNextDescribe } from 'e2e-utils'
+import path from 'path'
+import fs from 'fs-extra'
+
+const testedExamples = [
+  // Internal features
+  'active-class-name',
+  'amp',
+  'amp-first',
+  'amp-story',
+  'api-routes',
+  'api-routes-cors',
+  'api-routes-middleware',
+  'api-routes-rate-limit',
+  'api-routes-rest',
+  'app-dir-i18n-routing',
+  'app-dir-mdx',
+  'basic-css',
+  'basic-export',
+  'blog',
+  'blog-starter',
+  'catch-all-routes',
+  'custom-routes-proxying',
+  'custom-server',
+  'data-fetch',
+  'dynamic-routing',
+  'environment-variables',
+  'fast-refresh-demo',
+  'head-elements',
+  'headers',
+  'hello-world',
+  'hello-world-esm',
+  'i18n-routing',
+  'image-component',
+  'image-legacy-component',
+  'layout-component',
+  'middleware',
+  'middleware-matcher',
+  'modularize-imports',
+  'nested-components',
+  'next-css',
+  'next-forms',
+  'progressive-render',
+  'redirects',
+  'remove-console',
+  'reproduction-template',
+  'rewrites',
+  'script-component',
+  'ssr-caching',
+  'styled-jsx-with-csp',
+  'svg-components',
+  'using-router',
+  'with-absolute-imports',
+  'with-app-layout',
+  'with-context-api',
+  'with-env-from-next-config-js',
+  'with-loading',
+  'with-shallow-routing',
+  'with-sitemap',
+  'with-typescript',
+  'with-typescript-types',
+  'with-web-worker',
+  'with-webassembly',
+
+  // Library integrations that we can't break
+  'with-jest',
+  'with-jest-babel',
+  'with-mdx',
+  'with-mdx-remote',
+  'with-tailwindcss',
+  'with-turbopack',
+  'with-vercel-fetch',
+]
+
+const testExample = (example) => {
+  const exampleFiles = path.join(__dirname, '..', '..', 'examples', example)
+
+  const packageJson = fs.readJsonSync(path.join(exampleFiles, 'package.json'))
+  describe(`example '${example}'`, () => {
+    // If there is an issue during a build, jest won't tell us which example caused it
+    // we need to log it ourselfs
+    beforeAll(() => {
+      require('console').log(`Running example '${example}'`)
+    })
+    createNextDescribe(
+      `example '${example}'`,
```

Comment:
Is the `describe` duplicated on line 79 and here on line 86?

Reply:
Yes, See the comment. We need to fire `beforeAll` before we fire `createNextDescribe`, because that's where the actual building happens (and we want to log which example failed).